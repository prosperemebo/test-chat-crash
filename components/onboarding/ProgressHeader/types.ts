export interface ProgressHeaderProps {
    goBack: () => void
    steps: number,
    progress: number,
    completeSteps: number
}