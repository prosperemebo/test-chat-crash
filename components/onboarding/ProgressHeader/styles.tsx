import styled from 'styled-components/native'
import { Dimensions } from 'react-native'
import { Image } from 'expo-image'

const { width } = Dimensions.get('window')

type BarProps = {
  progress: number
}

export const Container = styled.View`
    margin-horizontal: 15px;
    margin-vertical: 5px;
    padding-horizontal: 5px;
`

export const Progress = styled.View`
  width: ${width * 0.9}px;
  height: 40px;
`

export const Header = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`

export const IconContainer = styled.TouchableOpacity`
  justify-content: center;
  align-items: center;
`

export const Icon = styled(Image)`
    height: 20px;
    width: 20px;
    resize-mode: contain;
`

export const ProgressLabel = styled.Text`
  font-family: 'Poppins-Bold';
  font-size: 17px;
  padding-left: 30px;
  color: ${(props) => props.theme.onboarding.headerTextColor};
`

export const LabelContainer = styled.View``

export const Label = styled.Text`
  font-style: normal;
  font-family: 'Poppins-Regular';
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  color: ${props => props.theme.onboarding.headerStepsTextColor};
`

export const Bar = styled.View`
  width: 100%;
  height: 5px;
  border-radius: 5px;
  background-color: ${props => props.theme.onboarding.progressBarBackgroundColor};
`
export const BarInner = styled.View<BarProps>`
  height: 100%;
  border-radius: 5px;
  width: ${(props) => (props.progress ? width * (props.progress / 100) * 0.9 : width * 0.9)}px;
  background-color: ${props => props.theme.onboarding.progressBarColor};
`