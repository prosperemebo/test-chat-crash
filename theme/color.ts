export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}