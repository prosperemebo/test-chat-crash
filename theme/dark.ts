import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
