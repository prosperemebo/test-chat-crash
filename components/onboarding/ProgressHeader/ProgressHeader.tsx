import i18n from '@/localization/i18n'
import { Container, Icon, IconContainer, Header, Progress, ProgressLabel, LabelContainer, Label, Bar, BarInner } from './styles'
import { ProgressHeaderProps } from './types'

const BackIcon = require('@/assets/images/onboarding/back_button.png')

const ProgressHeader = ({ goBack, steps, progress, completeSteps }: ProgressHeaderProps) => {
  const progressText = i18n.t('onb:progress', {
    stepProgress: completeSteps + 1,
    steps,
  }).toUpperCase()

  return <Container>
    <Progress>
      <Header>
        <IconContainer onPress={goBack}>
          <Icon size={2} source={BackIcon} />
        </IconContainer>
        <ProgressLabel>{i18n.t('login:register')}</ProgressLabel>
        <LabelContainer>
          <Label>
            {progressText}
          </Label>
        </LabelContainer>
      </Header>
      <Bar>
        <BarInner progress={progress} />
      </Bar>
    </Progress>
  </Container>
}


export default ProgressHeader