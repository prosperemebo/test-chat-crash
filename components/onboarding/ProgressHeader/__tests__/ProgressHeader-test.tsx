import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import ProgressHeader from '../'
import { dark } from '@/theme'

describe('<ProgressHeader />', () => {

  test('ProgressHeader renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <ProgressHeader goBack={() => {}} progress={3} steps={5} completeSteps={2}/>
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})