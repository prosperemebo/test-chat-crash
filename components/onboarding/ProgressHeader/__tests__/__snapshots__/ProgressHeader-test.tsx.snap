// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressHeader /> ProgressHeader renders correctly 1`] = `
<View
  style={
    {
      "marginHorizontal": 15,
      "marginVertical": 5,
      "paddingHorizontal": 5,
    }
  }
>
  <View
    style={
      {
        "height": 40,
        "width": 675,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          size={2}
          source={1}
          style={
            {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 17,
            "paddingLeft": 30,
          }
        }
      >
        MOCKED_STRING:login:register
      </Text>
      <View
        style={{}}
      >
        <Text
          style={
            {
              "color": "#8791a5",
              "fontFamily": "Poppins-Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          MOCKED_STRING:ONB:PROGRESS
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#8D88A8",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "height": 5,
          "width": "100%",
        }
      }
    >
      <View
        progress={3}
        style={
          {
            "backgroundColor": "#C47AF5",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": "100%",
            "width": 20.25,
          }
        }
      />
    </View>
  </View>
</View>
`;
